mentorship iosTests/HomeTests.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  HomeTests.swift
+//  Created on 26/07/20
+//  Created for AnitaB.org Mentorship-iOS 
+//
+
+import XCTest
+@testable import mentorship_ios
+
+class HomeTests: XCTestCase {
+    // custom urlsession for mock network calls
+    var urlSession: URLSession!
+
+    override func setUpWithError() throws {
+        // Set url session for mock networking
+        let configuration = URLSessionConfiguration.ephemeral
+        configuration.protocolClasses = [MockURLProtocol.self]
+        urlSession = URLSession(configuration: configuration)
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+    
+    // BUG - response always nil (print(resp))
+    func testHomeService() throws {
+        // Home Service
+        let homeService: HomeService = HomeAPI(urlSession: urlSession)
+
+        // Set mock json and data
+        let mockJSON = HomeModel.HomeResponseData(
+            asMentor: nil,
+            asMentee: RequestsList(sent: nil, received: nil),
+            tasksToDo: [TaskStructure(id: 0, description: "", isDone: true, createdAt: 0, completedAt: 0)],
+            tasksDone: [TaskStructure]()
+        )
+        let mockData = try JSONEncoder().encode(mockJSON)
+        
+        // Return data in mock request handler
+        MockURLProtocol.requestHandler = { request in
+            return (HTTPURLResponse(), mockData)
+        }
+        
+        // Set expectation. Used to test async code.
+        let expectation = XCTestExpectation(description: "response")
+        
+        // Make fetch dashboard request and test response data.
+        homeService.fetchDashboard { resp in
+            print(resp)
+            expectation.fulfill()
+        }
```

Comment:
One bug - 'resp' in completion handler contains nil values for all properties. Ideally, it should be equal to mockJSON declared in this function.

Reply:
@anitab-org/mentorship-ios-maintainers assert statements have been restored in the home service test (failing).